connector/connect/server/src/main/scala/org/apache/spark/sql/connect/service/SparkConnectService.scala

```diff
@@ -210,7 +211,12 @@ class SparkConnectService(debug: Boolean)
  * @param userId
  * @param session
  */
-case class SessionHolder(userId: String, sessionId: String, session: SparkSession)
+case class SessionHolder(
+  userId: String,
+  sessionId: String,
+  session: SparkSession,
+  mlCache: MLCache = MLCache()
```

Comment:
I feel we'd better let `SessionHolder` hold a generalized cache (`ObjectCache`) instead of a ML-specific cache

Reply:
But the modelCache has ml specific attribute, like `modelToHandlerMap`.

Reply:
https://github.com/apache/spark/pull/40485